packages/zoe/src/z2spec/escrow-exo.js

```diff
@@ -0,0 +1,311 @@
+/**
+ * @file escrow exchange using ownable seat, resolver
+ */
+// @ts-check
+
+import { AmountMath } from '@agoric/ertp';
+import { deeplyFulfilledObject } from '@agoric/internal';
+import { prepareDurablePublishKit } from '@agoric/notifier';
+import { assert, bare, Fail, q } from '@endo/errors';
+import { E } from '@endo/far';
+import { M, objectMap } from '@endo/patterns';
+import {
+  addToAllocation,
+  subtractFromAllocation,
+} from '../contractFacet/allocationMath.js';
+import { isOfferSafe } from '../contractFacet/offerSafety.js';
+import { prepareOwnable } from '../contractSupport/prepare-ownable.js';
+
+/**
+ * @import {Zone} from '@agoric/base-zone'
+ */
+
+const { keys, entries, fromEntries } = Object;
+const { isEmpty } = AmountMath;
+
+const objectFilter = (obj, pred) =>
+  fromEntries(entries(obj).filter(([k, v]) => pred(v, k)));
+
+/**
+ * @param {Zone} zone
+ * @param {<T>() => PublishKit<T>} makeDurablePublishKit
+ */
+export const prepareSeatKit = (zone, makeDurablePublishKit) =>
+  zone.exoClassKit(
+    'Seat Kit',
+    undefined,
+    proposal => {
+      /** @type {PublishKit<unknown>} */
+      const exitKit = makeDurablePublishKit();
+      /** @type {PublishKit<PaymentPKeywordRecord>} */
+      const payoutsKit = makeDurablePublishKit();
+      return {
+        proposal,
+        exitKit,
+        payoutsKit,
+        exiting: false,
+        /** @type {Allocation} */
+        currentAllocation: proposal.give,
+      };
+    },
+    {
+      readOnly: {
+        hasExited() {
+          return this.state.exiting;
+        },
+        getProposal() {
+          return this.state.proposal;
+        },
+        getCurrentAllocation() {
+          return this.state.currentAllocation;
+        },
+        getExitSubscriber() {
+          return this.state.exitKit.subscriber;
+        },
+      },
+
+      /** The right to collect payouts */
+      seat: {
+        /** @param {Keyword} kw */
+        async getPayout(kw) {
+          const { payoutsKit } = this.state;
+          const paid = await payoutsKit.subscriber.getUpdateSince();
+          return paid.value[kw];
+        },
+        async getPayouts() {
+          const { payoutsKit } = this.state;
+          const paid = await payoutsKit.subscriber.getUpdateSince();
+          return paid.value;
+        },
+
+        /** @param {unknown} completion */
+        async exit(completion) {
+          assert.fail('TODO');
+        },
+        fail(reason) {
+          assert.fail('TODO');
+        },
```

Comment:
Since these are only requests to exit, that may or may not exit depending on exit terms and the contract, and in any case will not cause a synchronous exit, we need to worry that it would still leave the seat in an exclusively transferable state, even though a seat-initialized exit may be in progress from (what would be) the previous owner.

Reply:
Maybe this is why your state variable is `exiting` rather than `exited`, so the seat becomes non-transferable as soon as an exit is requested from the seat? In which case

```suggestion
        async exit(completion) {
          this.seat.exiting = true;
          assert.fail('TODO');
        },
        fail(reason) {
          this.seat.exiting = true;
          assert.fail('TODO');
        },
```